testing/internal/e2e/tests/base/auth_unauthenticated_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package base_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hashicorp/boundary/testing/internal/e2e"
+	"github.com/hashicorp/boundary/testing/internal/e2e/boundary"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCliUnauthenticatedUserAccess(t *testing.T) {
+	e2e.MaybeSkipTest(t)
+
+	ctx := context.Background()
+	boundary.AuthenticateAdminCli(t, ctx)
+
+	// Check one authenticated request
+	output := e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"targets", "list",
+			"--recursive",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	// Logout
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"logout",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	// Attempt commands that unauthenticated users can make
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"scopes", "list",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"auth-methods", "list",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	// Check user cannot make requests that require authentication
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"targets", "list",
+			"--recursive",
+		),
+	)
+	require.Error(t, output.Err, "Unauthenticated user was able to access private resource")
```

Comment:
Do we want to include a check on the status code that is returned here? This is an example from another test
```
var response boundary.CliError
err = json.Unmarshal(output.Stderr, &response)
require.NoError(t, err)
require.Equal(t, http.StatusForbidden, int(response.Status))
```